src/include/migraphx/generate.hpp

```diff
@@ -36,19 +36,14 @@ inline namespace MIGRAPHX_INLINE_NS {
 template <class T, MIGRAPHX_REQUIRES(is_floating_point<T>{})>
 constexpr T normalize(unsigned long z)
 {
-    if(z == 0)
-        return T(0);
-    const auto max     = 32;
-    const double range = max / 2; // NOLINT
-    double result      = double(z % max) / range;
-    result -= 1;
-    return T(result);
+    const auto max = 1ULL << (sizeof(T) * 8 - 1);
+    return T(z % max);
```

Comment:
This makes all the numbers positive, and they are not between the values of (-1, 1).

Reply:
It wasn't clear what the expected output should be .. I will add a comment here as well. Thanks.

Reply:
Fixed.